src/Microsoft.AspNet.Mvc.Core/Controllers/FilterActionInvoker.cs

```diff
@@ -288,8 +289,18 @@ private Task InvokeAllAuthorizationFiltersAsync()
             var current = _cursor.GetNextFilter<IAuthorizationFilter, IAsyncAuthorizationFilter>();
             if (current.FilterAsync != null)
             {
+                _diagnosticSource.BeforeOnAuthorizationAsync(
+                    _authorizationContext.ActionDescriptor,
+                    _authorizationContext,
+                    current.FilterAsync);
```

Comment:
Simplify this method to just take the `_authorizationContext` and `current.FilterAsync`